virtool/groups/data.py

```diff
@@ -47,19 +52,37 @@ async def find(self) -> List[GroupMinimal]:
             async for document in self._mongo.groups.find()
         ]
 
-    async def get(self, group_id: str) -> Group:
+    async def get(self, group_id: str | int) -> Group:
```

Comment:
Looking closer, I think the new logic for getting the group should all be in `fetch_complete_group`. As written functions which that rely on `fetch_complete_group` will fail to grab data from postgres. You will have to modify the signature of `fetch_complete_group` to include postgres. While doing so, please work on streamlining the logic. I think there is significant room to streamline and reduce the number of `if` statements to increase readability.